metrics/network/remote_network/remote-networking-test-common.sh

```diff
@@ -0,0 +1,119 @@
+# tests that are using iperf
+
+# Name of the interface were swarm will run
+interface_name=$1
+# Name of the user (ssh)
+ssh_user=$2
+# Ip address (ssh)
+ssh_address=$3
+# Environment variable for docker
+DOCKER_EXE=docker
+# Image for network testing
+network_image="gabyct/network"
+# Number of replicas that will be launch in each host
+number_of_replicas=1
+# Timeout in seconds to verify replicas are running
+timeout=10
+# Name of the first service
+first_service="testswarm1"
+# Name of the second service
+second_service="testswarm2"
+# Port where the first service will run
+port_first_service="8080:80"
+# Port where the second service will run
+port_second_service="8081:80"
+
+# This function will start swarm as well as it will label
+# the nodes and create the replicas
+function setup {
+	token=$(mktemp)
+	$DOCKER_EXE swarm init --advertise-addr "$interface_name" > "$token"
+	token_name=$(cat "$token" | grep "token" | head -1 | cut -d '\' -f1)
+	ip_name=$(cat "$token" | grep ":" | tail -1)
+	ssh "$ssh_user"@"$ssh_address" "$DOCKER_EXE swarm join $token_name $ip_name" > /dev/null
+	first_node=$($DOCKER_EXE node ls -q | head -1)
+	second_node=$($DOCKER_EXE node ls -q | tail -1)
+	$DOCKER_EXE node update --label-add machine=machine1 "$first_node" > /dev/null
+	$DOCKER_EXE node update --label-add machine=machine2 "$second_node" > /dev/null
+	$DOCKER_EXE service create \
+		--name "$first_service" --replicas "$number_of_replicas" \
+		--constraint 'node.labels.machine == machine1' \
+		--publish "$port_first_service" "$network_image" sh > /dev/null
+	$DOCKER_EXE service create \
+		--name "second_service" --replicas "$number_of_replicas" \
+		--constraint 'node.labels.machine == machine2' \
+		--publish "$port_second_service" "$network_image" sh > /dev/null
+	rm -f "$token"
+}
+
+# This function will verify that the replica on the host client is running
+function client_replica_status {
+	for i in $(seq "$timeout"); do
+		client_status=$($DOCKER_EXE ps -q --filter=status=running | wc -l)
+		if [ "$client_status" -ge 1 ]; then
+			break
+		fi
+		sleep 1
+	done
+}
+
+# This function will verify that the replica on the host server is running
+function server_replica_status {
+	status=$(mktemp)
+	for i in $(seq "$timeout"); do
+		ssh "$ssh_user"@"$ssh_address" 'DOCKER_EXE=docker; \
+		$DOCKER_EXE ps -q --filter=status=running | wc -l' > "$status"
+		client_status=$(cat $status)
+		if [ "$client_status" -ge 1 ]; then
+			break
+		fi
+		sleep 1
+	done
+	rm -f "$status"
+}
+
+# This function will verify the ip address of the server
+function check_iperf3_server_address {
+	ssh "$ssh_user"@"$ssh_address" 'DOCKER_EXE=docker; \
+				network_name=$($DOCKER_EXE network ls --filter driver=overlay -q); \
+				$DOCKER_EXE network inspect "$network_name" --format "{{ range .Containers}} {{ println .IPv4Address}}{{end}}" | head -n -2 | cut -d'/' -f1'
+}
+
+# This function will start an iperf3 server
+function start_iperf3_server {
```

Comment:
maybe you should take a look in `lib/network-common.bash` I am not sure but some functions looks similar to some of them, it is possible that you can reuse those functions instead to rewrite them